File: android/src/main/java/org/maplibre/reactnative/components/mapview/helpers/CameraChangeTracker.java
package org.maplibre.reactnative.components.mapview.helpers;

public class CameraChangeTracker {
    public static final int USER_GESTURE = 1;
    public static final int DEVELOPER_ANIMATION = 2;
    public static final int SDK_ANIMATION = 3;
    public static final int EMPTY = -1;

    private int reason = EMPTY;
    public boolean isAnimating;

    public void setReason(int reason) {
        this.reason = reason;
    }


    public boolean isUserInteraction() {
        return reason == USER_GESTURE || reason == DEVELOPER_ANIMATION;
    }

    public boolean isAnimated() {
        return reason == DEVELOPER_ANIMATION || reason == SDK_ANIMATION;
    }


    public boolean isEmpty() {
        return reason == EMPTY;
    }
}

File: package/android/src/main/java/org/maplibre/reactnative/components/mapview/helpers/CameraChangeTracker.kt
package org.maplibre.reactnative.components.mapview.helpers

class CameraChangeTracker {
    private var reason: Int = EMPTY
    var isAnimating: Boolean = false

    fun setReason(reason: Int) {
        this.reason = reason
    }

    val isUserInteraction: Boolean
        get() = reason == USER_GESTURE || reason == DEVELOPER_ANIMATION

    val isAnimated: Boolean
        get() = reason == DEVELOPER_ANIMATION || reason == SDK_ANIMATION

    val isEmpty: Boolean
        get() = reason == EMPTY

    companion object {
        const val EMPTY: Int = -1
        const val USER_GESTURE: Int = 1
        const val DEVELOPER_ANIMATION: Int = 2
        const val SDK_ANIMATION: Int = 3
    }
}
